tests/chart_tests/test_default_chart.py

```diff
@@ -12,6 +12,38 @@
 pod_managers = ["Deployment", "StatefulSet", "DaemonSet"]
 
 
+class TestK8sVersionConstraints:
+    @pytest.mark.parametrize(
+        "version,err_substring",
+        [
+            (k8s_version_too_old, "too old!"),
+            (k8s_version_too_new, "too new!"),
+        ],
+    )
+    def test_k8s_version_out_of_bounds(self, version, err_substring):
+        """Test that an error is returned when the k8s version is too old or too new."""
+        with pytest.raises(Exception) as err:
+            render_chart(
+                kube_version=version,
+            )
+        assert err_substring in str(err.value.stderr.decode())
+
+    @pytest.mark.parametrize(
+        "version",
+        [
+            (k8s_version_too_old),
+            (k8s_version_too_new),
+        ],
+    )
+    def test_k8s_version_out_of_bounds_override(self, version):
+        """Test that no error is returned for versions that are too old or new when forceIncompatibleKubernetes is used."""
+        render_chart(
+            values={"forceIncompatibleKubernetes": True},
```

Comment:
@danielhoherd @rishkarajgi option forceIncompatibleKubernetes is not in our config.yml we should add it in defaults

Reply:
@pgvishnuram I guess best practice would be to add it in values.yaml, but it's supposed to be something customers don't use, so I'd rather not bring attention to it.